vmaas/load.go

```diff
@@ -1086,3 +1089,86 @@ func loadCpeID2Label(c *Cache) {
 	}
 	c.CpeID2Label = ret
 }
+
+func loadCSAFProductStatus(c *Cache) {
+	now := time.Now()
+
+	cnt := getCount("csaf_product_status", "*")
+	rows := getAllRows("csaf_product_status", "id,name")
+
+	cache := make(map[int]string, cnt)
+
+	type ProductStatusRow struct {
+		ID    int
+		label string
+	}
+	psr := ProductStatusRow{}
+
+	for rows.Next() {
+		if err := rows.Scan(&psr.ID, &psr.label); err != nil {
+			utils.TimeTrack(now, "CSAF product status")
+			log.Fatal("failed to scan csaf_product_status row", err)
+		}
+		cache[psr.ID] = psr.label
+	}
+
+	c.CSAFProductStatus = cache
+	utils.TimeTrack(time.Now(), "CpeID2Label")
+}
+
+func loadCSAFCVE(c *Cache) {
+	now := time.Now()
+
+	cnt := getCount("csaf_cve_product", "*")
+	rows := getAllRows("csaf_cve_product", "id,cve_id,csaf_product_id,csaf_product_status_id")
+
+	csafCVEProduct := make(map[CSAFProductID][]CSAFCVEProduct, cnt)
+	ccp := CSAFCVEProduct{}
+
+	for rows.Next() {
+		if err := rows.Scan(&ccp.ID, &ccp.CVEID, &ccp.CSAFProductID, &ccp.CSAFProductStatusID); err != nil {
+			utils.TimeTrack(now, "CSAF CVEs")
+			log.Fatal("failed to scan csaf_cve_product row", err)
+		}
+		csafCVEProduct[ccp.CSAFProductID] = append(csafCVEProduct[ccp.CSAFProductID], ccp)
+	}
+
+	rows = getAllRows("csaf_product", "id,cpe_id,package_name_id,package_id,module_stream_id")
+	cnt = getCount("csaf_product", "*")
+
+	cveCache := make(map[CSAFProduct][]CVEID, cnt)
+
+	type CSAFProductRow struct {
+		ID      int
+		Product CSAFProduct
+	}
+
+	cpr := CSAFProductRow{}
+	for rows.Next() {
+		if err := rows.Scan(&cpr.ID,
+			&cpr.Product.CpeID,
+			&cpr.Product.PackageNameID,
+			&cpr.Product.PackageID,
+			&cpr.Product.ModuleStream); err != nil {
+			utils.TimeTrack(now, "CSAF CVEs")
+			log.Fatal("failed to scan csaf_product row", err)
+		}
+
+		cveProducts := csafCVEProduct[CSAFProductID(cpr.ID)]
+		cves := make([]CVEID, len(cveProducts))
```

Comment:
this looks like the reason for those CVEs with id=0, append is used below so we need to set the slice length to 0
```suggestion
		cves := make([]CVEID, 0, len(cveProducts))
```